pkg/controller/migcluster/storageclass.go

```diff
@@ -0,0 +1,36 @@
+package migcluster
+
+import (
+	"strconv"
+
+	migapi "github.com/fusor/mig-controller/pkg/apis/migration/v1alpha1"
+	k8sclient "sigs.k8s.io/controller-runtime/pkg/client"
+)
+
+// Sets the list of storage classes from the cluster
+func (r *ReconcileMigCluster) setStorageClasses(cluster *migapi.MigCluster) error {
+	var client k8sclient.Client
+	client, err := cluster.GetClient(r)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+	clusterStorageClasses, err := cluster.GetStorageClasses(client)
+	if err != nil {
+		log.Trace(err)
+		return err
+	}
+	var storageClasses []migapi.StorageClass
+	for _, clusterStorageClass := range clusterStorageClasses {
+		storageClass := migapi.StorageClass{
+			Name:        clusterStorageClass.Name,
+			Provisioner: clusterStorageClass.Provisioner,
+		}
+		if clusterStorageClass.Annotations != nil {
+			storageClass.Default, _ = strconv.ParseBool(clusterStorageClass.Annotations["storageclass.kubernetes.io/is-default-class"])
+		}
+		storageClasses = append(storageClasses, storageClass)
+		cluster.Spec.StorageClasses = storageClasses
```

Comment:
I think you want this outside the `for` loop.  Right?
This will only update the `Spec` when `clusterStorageClasses` isn't empty.  If there are no storage classes on the cluster, shouldn't the `Spec.StorageClasses` reflect that?

Reply:
Good catch. Yes, this line needs to be moved outside the loop.